mitodl/micromasters
# Export exam authorizations to Pearson

#### What are the relevant tickets?

Fixes #2076 

#### What's this PR do?

This PR implements exam authorization for Pearson. This involves generating a TSV file and uploading it to their FTP server. This builds on TSV writer and SFTP upload code written in PR #2060 for issue #1795.

Also cleaned up some of the tests around existing Pearson code so they're more thorough / legible.

If you review this, please ping me and I'll share the detailed docs on their API (I believe they're confidential so can't be shared here).

#### Where should the reviewer start?

exams/pearson.py

#### How should this be manually tested?

In a Django shell:

```python
    from factory.django import mute_signals
    from django.db.models.signals import post_save
    from exams.factories import ExamProfileFactory
    from exams.tasks import export_exam_authorizations
    OFFSET=1200
    with mute_signals(post_save):
        for x in range(50):
            profile = ProfileFactory.create(user____sequence=OFFSET+x)
            ExamAuthorizationFactory.create(user=profile.user)
    export_exam_authorizations()
```

Then in another terminal session session: `sftp -P 2022 odl@localhost:share` (password: 123).

You can `ls` to see the file that was exported and use `get` to download it to your host working directory so you can inspect the contents.
